unit_test/degrib2c.c

```diff
@@ -0,0 +1,168 @@
+/** @file
+ *  @author Vuong @date 2021-02-05
+ *
+ *   This program tests G2C library by calling number of routines
+ *   to read GRIB2 file and makes an inventory of GRIB2 file.
+ *   The inventory will displays: 
+ *   -SECTION 0, 1
+ *   -Grid Definition in section 3 (GDS)
+ *   -Product Definition Templates (PDT) used in Section 4
+ *   -Data Representation Section (DRS) in Section 5
+ *   -Data Values in Data section
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include "grib2.h"
+
+int main(int argc, char *argv[])
+{
+      unsigned char *cgrib=0;
+      g2int  listsec0[3],listsec1[13];
+      g2int numlocal,numfields;
+      g2int currlen=0,itot,lskip,j,n,icount,lgrib,iseek;
+      g2int unpack,expand,ret,ierr;
+      g2float fldmax,fldmin,sum;
+      gribfield  *gfld;
+      FILE  *ifl1;
+      size_t  lengrib;
+
+      g2int  msk1=32000;
+
+      //start();
+      unpack=1;
+      expand=0;
+      
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+//  GET ARGUMENTS
+      if (argc != 2) {
+        printf("degrib2c:  Incorrect usage\n");
+        printf("Usage: degrib2c grib2file\n");
+        return 2;
+      }
+
+      ifl1=fopen(argv[1],"r");
+      if ( ifl1 == 0 ) {
+         perror("cannot open file:");
+         return 3;
+      }
+
+      itot=0;
+      icount=0;
+      iseek=0;
+
+      for (;;) {
+         seekgb(ifl1,iseek,msk1,&lskip,&lgrib);
+         if (lgrib == 0) break;    // end loop at EOF or problem
+         if (lgrib > currlen) {
+            if (cgrib!=0) free(cgrib);
+            cgrib=(unsigned char *)malloc(lgrib);
+            currlen=lgrib;
+         }
+         //call baread(ifl1,lskip,lgrib,lengrib,cgrib)
+         ret=fseek(ifl1,lskip,SEEK_SET);
+         lengrib=fread(cgrib,sizeof(unsigned char),lgrib,ifl1);
+         if (lgrib != lengrib) {
+            printf(" degrib2c: IO Error.\n");
+            return 9;
+         }
+         iseek=lskip+lgrib;
+         icount++;
+
+         printf("\n GRIB MESSAGE  %ld starts at %ld \n\n",icount,lskip+1);
+
+// Unpack GRIB2 field
+         ierr=g2_info(cgrib,listsec0,listsec1,&numfields,&numlocal);
+         if (ierr != 0) {
+           printf(" ERROR querying GRIB2 message = %d\n",ierr);
+           return 10;
+         }
+         itot=itot+numfields;
+         printf("  SECTION 0:  ");
+         for (j=0;j<3;j++) printf("%ld ",listsec0[j]);
+         printf("\n");
+         printf("  SECTION 1:  ");
+         for (j=0;j<13;j++) printf("%ld ",listsec1[j]);
+         printf("\n");
+         printf("  Contains  %ld  Local Sections  and  %ld  data fields.\n",
```

Comment:
Instead of printing out these values, check them against constants in the program which represent the expected values for the data file(s) you've chosen.

Tests must run without supervision - we cannot count on anyone looking at the output to determine it's correct.